### PR TITLE
PIA-634: Android Logout Add e2e Coverage

### PR DESCRIPTION
**Pull request by kp-ruth-ondrada**
## Purpose
This PR is to add logout e2e tests to our code. It covers checking for 2 important features we have for PIA, which is Dedicated IP and Protocol selection.

The expected results is that once user has logout from the android mobile application and log back in again, the previously selected protocol that is not set by default will no longer be selected. Same goes for Dedicated IP, the DIP entered by the user once logout will be removed.

### CODE REVIEW

File: app/src/androidTest/java/screens/steps/SideMenuStepObjects.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import com.privateinternetaccess.android.screens.objects.SideMenuPageObjects
+
+class SideMenuStepObjects {
+
+    private val sideMenuPageObjects = SideMenuPageObjects()
+
+    fun clickOnDedicatedIP(){
+        sideMenuPageObjects.dedicateIP.click()
+    }
```
Comment (kp-gabriel-sasu) [inline app/src/androidTest/java/screens/steps/SideMenuStepObjects.kt:11]:
```suggestion
    }
    
```

File: app/src/androidTest/java/tests/SignInTests.kt [hunk 2]
```diff
@@ -1,63 +1,74 @@
...
+                    {"username": "user", "password": "pass"},
+                    {"username": "", "password": ""},
+                    {"username": "${BuildConfig.PIA_VALID_USERNAME}", "password": ""},
+                    {"username": "", "password": "${BuildConfig.PIA_VALID_PASSWORD}"}
+                ]
+            """
+            val testData = DataFactory.loadTestDataFromJson(jsonTestData)
+            return testData.map { arrayOf(it.username, it.password) }
+        }
     }
```
Comment (kp-gabriel-sasu) [inline app/src/androidTest/java/tests/SignInTests.kt]: Assuming you move the json data to an external file held in a `data` folder this whole test can be improved in terms of efficiency by doing the following

```json
[
    {"username": "PLACEHOLDER_VALID_USERNAME", "password": "PLACEHOLDER_VALID_PASSWORD", "expectedOutcome": "valid"},
    {"username": "user", "password": "pass", "expectedOutcome": "invalid"},
    {"username": "", "password": "", "expectedOutcome": "noCredentials"},
    {"username": "PLACEHOLDER_VALID_USERNAME", "password": "", "expectedOutcome": "invalid"},
    {"username": "", "password": "PLACEHOLDER_VALID_PASSWORD", "expectedOutcome": "invalid"}
]
```

Data function changes:
```kt
fun data() : Collection<Array<String>> {
    val jsonTestData = File("path_to_test_data.json").readText()
        .replace("PLACEHOLDER_VALID_USERNAME", BuildConfig.PIA_VALID_USERNAME)
        .replace("PLACEHOLDER_VALID_PASSWORD", BuildConfig.PIA_VALID_PASSWORD)
    
    val testData = DataFactory.loadTestDataFromJson(jsonTestData)
    return testData.map { arrayOf(it.username, it.password, it.expectedOutcome) }
}
```

File: app/src/androidTest/java/tests/SignInTests.kt [hunk 3]
```diff
@@ -1,63 +1,74 @@
...
+        if (username == BuildConfig.PIA_VALID_USERNAME && password == BuildConfig.PIA_VALID_PASSWORD) {
+            Log.i("Login Test", "Test with valid credentials (username: ${username} and password: ${password}) passed", Throwable("Test failed"))
+            assert(MainScreenPageObjects().connectButton.exists())
+        } else if (this.username == "user" && password == "pass") {
+            Log.i("Login Test", "Test with invalid credentials (username: ${username} and password: ${password}) passed", Throwable("Test failed"))
+            assert(SignInPageObjects().reachLoginScreenButton.exists())
+        } else {
+            Log.i("Login Test", "Test without credentials (username: ${username} and password: ${password}) passed", Throwable("Test failed"))
+            assert(SignInPageObjects().noUsernameOrPasswordError.exists())
+            }
```
Comment (kp-gabriel-sasu) [inline app/src/androidTest/java/tests/SignInTests.kt]: With the assumption in mind that the JSON changes suggested above are applied where the objects receive an addition key/value pair of `expectedOutcome:` the test and assertions can be improved with the following logic and less impact on performance and memory usage

```kt
@Test
fun loginDataDrivenTests() {
    stepObjects.allowNotifications()
    stepObjects.reachSignInScreen()
    stepObjects.enterCredentials(username, password)
    stepObjects.clickOnLoginButton()
    stepObjects.allowVpnProfileCreation()

    when (expectedOutcome) {
        "valid" -> {
            Log.i("Login Test", "Test with valid credentials passed")
            assert(MainScreenPageObjects().connectButton.exists())
        }
        "invalid" -> {
            Log.i("Login Test", "Test with invalid credentials passed")
            assert(SignInPageObjects().reachLoginScreenButton.exists())
        }
        "noCredentials" -> {
            Log.i("Login Test", "Test without credentials passed")
            assert(SignInPageObjects().noUsernameOrPasswordError.exists())
        }
    }
}
```

File: app/src/androidTest/java/tests/SignOutTests.kt [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        returnOnMainScreen()
+        successfulLogout()
+        successfulLogin()
+        goToSideMenu(sideMenuPageObjects.dedicateIP)
+        assert(DedicateIPPageObjects().dedicateIPField.text.equals("Paste your token here"))
+        assert(!DedicateIPPageObjects().serverFlag.exists())
+        assert(!DedicateIPPageObjects().serverName.exists())
+        assert(!DedicateIPPageObjects().serverIPAddress.exists())
+
+    }
```
Comment (kp-gabriel-sasu) [inline app/src/androidTest/java/tests/SignOutTests.kt]: There are multiple ways to optimise this code. It's not really a best practice to have multiple assertions in a single scenarios as it makes it hard to understand what exactly is being tested. Additionally with hard assertions you end up in situations where if the first assertion fails and stops your tests, you won't know the result of the second assertion until the first one is fixed, and the same repeating process applies for the subsequent assertions.

We are currently investigating the adoption of `AssertJ` assertion library for Kotlin projects which allows for the implementation of soft assertions that in turn provide the expected behaviour of going through all assertions and failing you scenario correctly without causing the bottleneck mentioned earlier. Until the research has an outcome on the adoption of `AssertJ` you can opt for one of the following approaches for temporary improvements.

Option 1 - Single assertion of an Object (slightly dirty but gets the job done)
```kt
@Test
fun validateDIPSettingsAtLogout() {
    successfulLogin()
    goToSideMenu(sideMenuPageObjects.dedicateIP)
    dedicateIPStepObjects.enterDedicatedIP()
    returnOnMainScreen()
    successfulLogout()
    successfulLogin()
    goToSideMenu(sideMenuPageObjects.dedicateIP)
    
    val isDIPSettingsDefault = checkDIPSettingsDefault()
    assert(isDIPSettingsDefault)
}

// This following function can sit in a Page Object or Step Object
fun checkDIPSettingsDefault(): Boolean {
    val dipObjects = DedicateIPPageObjects()
    return dipObjects.dedicateIPField.text.equals("Paste your token here") &&
           !dipObjects.serverFlag.exists() &&
           !dipObjects.serverName.exists() &&
           !dipObjects.serverIPAddress.exists()
}
```

Option 2 - Looping through multiple assertions

```kt
@Test
fun validateDIPSettingsAtLogout() {
    successfulLogin()
    goToSideMenu(sideMenuPageObjects.dedicateIP)
    dedicateIPStepObjects.enterDedicatedIP()
    returnOnMainScreen()
    successfulLogout()
    successfulLogin()
    goToSideMenu(sideMenuPageObjects.dedicateIP)
    
    val conditions = listOf(
        { DedicateIPPageObjects().dedicateIPField.text.equals("Paste your token here") },
        { !DedicateIPPageObjects().serverFlag.exists() },
        { !DedicateIPPageObjects().serverName.exists() },
        { !DedicateIPPageObjects().serverIPAddress.exists() }
    )

    for (condition in conditions) {
        assert(condition())
    }
}

```

Comment (kp-ruth-ondrada) [reply app/src/androidTest/java/tests/SignOutTests.kt]: thank you!  went with option 2 🙏

File: app/src/androidTest/java/tests/SignInTests.kt (hunk 2 above)
Comment (kp-ruth-ondrada) [reply app/src/androidTest/java/tests/SignInTests.kt]: Would it be possible to have this change in another ticket? This code isn't written by me but would like to understand what's happening and prefer to work on it in isolation together with @kp-andrei-madescu.

File: app/src/androidTest/java/tests/SignInTests.kt (hunk 3 above)
Comment (kp-ruth-ondrada) [reply app/src/androidTest/java/tests/SignInTests.kt]: same as above 🙏

**Comment by kp-aldo-pedromingo** [comment]
Commits:
a7bf33c050fb8809260a158e01097280eb6ffeb0
and
d80a119df8df61d0eef59011837134c7c38623fb
appear to be partially signed. This might be an issue, as the PR sniffer will block the merge of the PR.

@kp-andrei-madescu it looks like your commits are not being correctly signed?

A solution for this could be to do a rebase and squash the faulty commits into another.

File: README.MD [hunk 5]
```diff
@@ -73,6 +73,11 @@ For more details please see [CONTRIBUTING](/CONTRIBUTING.md).
 
 Issues and Pull Requests should use these templates: [ISSUE](/.github/ISSUE_TEMPLATE.md) and [PULL REQUEST](/.github/PULL_REQUEST_TEMPLATE.md).
 
+### Testing
+
+In order to run the instrumented test we need to set environment variables containing valid credentials.
```
Comment (kp-aldo-pedromingo) [inline README.MD:78]: I'd replace `instrumented` for `integration`. Then change the folder name where the tests are contained to be more descriptive. In this case `src/integrationTests/java`

File: README.MD [hunk 6]
```diff
@@ -73,6 +73,11 @@ For more details please see [CONTRIBUTING](/CONTRIBUTING.md).
...
+They need to have the format `PIA_VALID_USERNAME` and `PIA_VALID_PASSWORD`.
```
Comment (kp-aldo-pedromingo) [inline README.MD]: Should we specify the other environment variable `PIA_VALID_DIP_TOKEN`?

File: app/src/androidTest/java/tests/SignInTests.kt [hunk 7]
```diff
@@ -1,63 +1,74 @@
...
+                """
+                [
+                    {"username": "${BuildConfig.PIA_VALID_USERNAME}", "password": "${BuildConfig.PIA_VALID_PASSWORD}"},
+                    {"username": "user", "password": "pass"},
+                    {"username": "", "password": ""},
+                    {"username": "${BuildConfig.PIA_VALID_USERNAME}", "password": ""},
+                    {"username": "", "password": "${BuildConfig.PIA_VALID_PASSWORD}"}
+                ]
+            """
+            val testData = DataFactory.loadTestDataFromJson(jsonTestData)
```
Comment (kp-aldo-pedromingo) [inline app/src/androidTest/java/tests/SignInTests.kt:40]: Hmmm, we don't really need to parse a json here. You can simply have an array of `TestData` and use that directly.

File: app/src/androidTest/java/helpers/ActionHelpers.kt [hunk 8]
```diff
@@ -2,18 +2,54 @@ package com.privateinternetaccess.android.helpers
...
         }
     }
 
     fun <T> inputTextInField(field: UiObject, data: T? = null) {
-        field.clearTextField()
         field.click()
         field.text = data?.toString() ?: ""
     }
+
+    fun successfulLogin() {
```
Comment (kp-aldo-pedromingo) [inline app/src/androidTest/java/helpers/ActionHelpers.kt:31]: Sweeeet! Very clean tests.

File: app/src/androidTest/java/helpers/LocatorHelper.kt [hunk 9]
```diff
@@ -12,4 +12,12 @@ object LocatorHelper {
     fun findByResourceId(id: String, instance: Int = 0): UiObject {
         return device.findObject(UiSelector().resourceId(id).instance(instance))
     }
+
+    fun findByContentDesc(description: String): UiObject {
```
Comment (kp-aldo-pedromingo) [inline app/src/androidTest/java/helpers/LocatorHelper.kt:16]: Not sure I really like this function and `findByText` because `ResourceIds` should be long lived, but `Content Descriptions` and `Text` might change more often, leading to tests throwing false negative failures when these descriptors are changed.

Are we able to add `ResourceIDs` to the elements which are being found by functions `findByContentDesc` and `findByText`? Not sure if this is feasible 🤔

File: app/src/androidTest/java/screens/objects/ProtocolPageObjects.kt [hunk 10]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.privateinternetaccess.android.helpers.LocatorHelper
+
+class ProtocolPageObjects {
+
+    val protocolSettings = LocatorHelper.findByResourceId("com.privateinternetaccess.android:id/protocolsSettings")
+    val protocolSelection = LocatorHelper.findByResourceId("com.privateinternetaccess.android:id/protocolSetting")
+    val openVPN = LocatorHelper.findByText("OpenVPN")
+    val wireGuard = LocatorHelper.findByText("WireGuard")
+    val save = LocatorHelper.findByResourceId("android:id/button1")
```
Comment (kp-aldo-pedromingo) [inline app/src/androidTest/java/screens/objects/ProtocolPageObjects.kt:11]: `button1` does not seem like a great ID 😄 We could refactor it to a more meaningful name while we are at it

File: app/src/androidTest/java/screens/objects/SideMenuPageObjects.kt [hunk 11]
```diff
@@ -0,0 +1,11 @@
+package com.privateinternetaccess.android.screens.objects
+
+import com.privateinternetaccess.android.helpers.LocatorHelper
+
+class SideMenuPageObjects {
+
+    val dedicateIP = LocatorHelper.findByText("Dedicated IP")
+    val settings = LocatorHelper.findByText("Settings")
+    val logout = LocatorHelper.findByText("Log out")
+    val logoutConfirm = LocatorHelper.findByResourceId("android:id/button1")
```
Comment (kp-aldo-pedromingo) [inline app/src/androidTest/java/screens/objects/SideMenuPageObjects.kt:10]: `button1`. Boy scout rule 😄

File: app/src/androidTest/java/screens/steps/DedicateIPStepObjects.kt [hunk 12]
```diff
@@ -0,0 +1,18 @@
+package com.privateinternetaccess.android.screens.steps
+
+import com.privateinternetaccess.android.BuildConfig
+import com.privateinternetaccess.android.helpers.ActionHelpers.inputTextInField
+import com.privateinternetaccess.android.screens.objects.DedicateIPPageObjects
+import com.privateinternetaccess.android.core.BaseUiAutomatorClass.Companion.defaultTimeOut
+
+
+class DedicateIPStepObjects {
```
Comment (kp-aldo-pedromingo) [inline app/src/androidTest/java/screens/steps/DedicateIPStepObjects.kt]: Refactor to `DedicatedIPStepObjects`. This applies also to `dedicatedIPPageObjects`

File: app/src/androidTest/java/screens/objects/DedicateIPPageObjects.kt [hunk 13]
```diff
@@ -0,0 +1,11 @@
+package com.privateinternetaccess.android.screens.objects
+
+import com.privateinternetaccess.android.helpers.LocatorHelper
+
+class DedicateIPPageObjects {
```
Comment (kp-aldo-pedromingo) [inline app/src/androidTest/java/screens/objects/DedicateIPPageObjects.kt]: Refactor to `DedicatedIPPageObjects`
Also `dedicatedIPField` 😄

**Review by kp-aldo-pedromingo — COMMENTED** [review]
Some minor things here and there. But overall it looks great. Good job! 💪

File: README.MD (hunk 5 above)
Comment (kp-ruth-ondrada) [reply README.MD:78]: We will address this to another ticket as per conversation with Gabriel 🙏 (see above)

File: README.MD (hunk 6 above)
Comment (kp-ruth-ondrada) [reply README.MD]: We will address this to another ticket as per conversation with Gabriel 🙏 (see above)

File: app/src/androidTest/java/tests/SignInTests.kt (hunk 7 above)
Comment (kp-ruth-ondrada) [reply app/src/androidTest/java/tests/SignInTests.kt:40]: We will address this to another ticket as per conversation with Gabriel 🙏 (see above)

File: app/src/androidTest/java/helpers/LocatorHelper.kt (hunk 9 above)
Comment (kp-ruth-ondrada) [reply app/src/androidTest/java/helpers/LocatorHelper.kt:16]: I don't know how to add resource id's yet, and would rather do it in another ticket and leave this right now.

File: app/src/androidTest/java/screens/objects/ProtocolPageObjects.kt (hunk 10 above)
Comment (kp-ruth-ondrada) [reply app/src/androidTest/java/screens/objects/ProtocolPageObjects.kt:11]: same as above

File: app/src/androidTest/java/screens/objects/SideMenuPageObjects.kt (hunk 11 above)
Comment (kp-ruth-ondrada) [reply app/src/androidTest/java/screens/objects/SideMenuPageObjects.kt:10]: same as above

File: README.MD (hunk 5 above)
Comment (kp-gabriel-sasu) [reply README.MD:78]: Hey @kp-aldo-pedromingo these are not integration testes therefore renaming this to `integration` would be inaccurate. Additionally the folder naming is Java/JVM/Kotlin convention based on targets and project setup. Renaming the folder structure would yield more effort and would impact the project setup

File: app/src/androidTest/java/tests/SignInTests.kt (hunk 7 above)
Comment (kp-gabriel-sasu) [reply app/src/androidTest/java/tests/SignInTests.kt:40]: @kp-aldo-pedromingo JSON usage for data driven testing is a common and industry standard approach. The alternatives would be databases, cloud providers. The only other suitable alternatives would be to build custom data providers however there are no real advantages to this since data providers work similarly with external sources such as JSON, XMLs etc and would in turn result in the same behaviour and value

File: app/src/androidTest/java/helpers/LocatorHelper.kt (hunk 9 above)
Comment (kp-gabriel-sasu) [reply app/src/androidTest/java/helpers/LocatorHelper.kt:16]: @kp-aldo-pedromingo I agree that ResourceIDs are better if not the best option. However since the app will be replaced with a newer implementation, we could discuss implementing a standard and naming convention as well as a mandatory addition of resourceIds. Right now it would be relatively redundant and time consuming to do this for all elements that currently do not have them.

File: app/src/androidTest/java/screens/objects/ProtocolPageObjects.kt (hunk 10 above)
Comment (kp-gabriel-sasu) [reply app/src/androidTest/java/screens/objects/ProtocolPageObjects.kt:11]: @kp-aldo-pedromingo These are Android Native IDs. We can not change their values, information, meta or anything related to them since they come natively from the OS

File: app/src/androidTest/java/screens/objects/SideMenuPageObjects.kt (hunk 11 above)
Comment (kp-gabriel-sasu) [reply app/src/androidTest/java/screens/objects/SideMenuPageObjects.kt:10]: @kp-aldo-pedromingo These are Android Native IDs. We can not change their values, information, meta or anything related to them since they come natively from the OS

File: app/src/androidTest/java/screens/objects/ProtocolPageObjects.kt (hunk 10 above)
Comment (kp-juan-docal) [reply app/src/androidTest/java/screens/objects/ProtocolPageObjects.kt:11]: @kp-gabriel-sasu +1 on Gabriel's comment. These are ids we cannot change as they are part of Android's AlertDialog

File: README.MD (hunk 5 above)
Comment (kp-aldo-pedromingo) [reply README.MD:78]: Got it! Just learned about instrumented tests, let's keep it as it is then.

File: app/src/androidTest/java/helpers/LocatorHelper.kt (hunk 9 above)
Comment (kp-aldo-pedromingo) [reply app/src/androidTest/java/helpers/LocatorHelper.kt:16]: We can just add it to the elements that are being targeted by `findByContentDesc` and `findByText` which shouldn't be much.

We can add the rest as we make more tests.

Comment (kp-juan-docal) [reply app/src/androidTest/java/helpers/LocatorHelper.kt:16]: My two cents here. @kp-aldo-pedromingo there are items that we don't control. For instance, everything related to the side menu (burger menu), and Android objects as I've mentioned in another comment below. Meaning that we cannot set those `resourceIds` without re-implementing the whole feature to something we fully control. I understand it is not ideal to not use `resourceIds`, but sometimes we simply have to avoid re-inventing an object for the sake of adding an id.

File: app/src/androidTest/java/screens/objects/ProtocolPageObjects.kt (hunk 10 above)
Comment (kp-aldo-pedromingo) [reply app/src/androidTest/java/screens/objects/ProtocolPageObjects.kt:11]: Ohh, that's right! I guess there is no alternative to this 😢

File: app/src/androidTest/java/tests/SignInTests.kt (hunk 7 above)
Comment (kp-aldo-pedromingo) [reply app/src/androidTest/java/tests/SignInTests.kt:40]: If we are moving the test data to a file this approach makes sense.

However, if we plan on keeping the test data inside the test source code, then we can simplify this by declaring and building the array of tuples in one step. Something like:

```
return testData.map { arrayOf(
     "${BuildConfig.PIA_VALID_USERNAME}" to "${BuildConfig.PIA_VALID_PASSWORD}", 
     "user" to "pass",
     ...
     )
```

We can then remove the call to `loadTestDataFromJson` and its dependencies.

File: app/src/androidTest/java/helpers/LocatorHelper.kt (hunk 9 above)
Comment (kp-aldo-pedromingo) [reply app/src/androidTest/java/helpers/LocatorHelper.kt:16]: I see 🤔 Thanks for the explanation @kp-juan-docal !

File: app/src/androidTest/java/tests/SignInTests.kt (hunk 2 above)
Comment (kp-ruth-ondrada) [reply app/src/androidTest/java/tests/SignInTests.kt]: We've fixed this!

File: app/src/androidTest/java/tests/SignInTests.kt (hunk 3 above)
Comment (kp-ruth-ondrada) [reply app/src/androidTest/java/tests/SignInTests.kt]: we've fixed this!

File: README.MD (hunk 6 above)
Comment (kp-ruth-ondrada) [reply README.MD]: fixed this!

**Comment by kp-andrei-madescu** [comment]
> Commits: [a7bf33c](https://github.com/pia-foss/mobile-android/commit/a7bf33c050fb8809260a158e01097280eb6ffeb0) and [d80a119](https://github.com/pia-foss/mobile-android/commit/d80a119df8df61d0eef59011837134c7c38623fb) appear to be partially signed. This might be an issue, as the PR sniffer will block the merge of the PR.
> 
> @kp-andrei-madescu it looks like your commits are not being correctly signed?
> 
> A solution for this could be to do a rebase and squash the faulty commits into another.

Not sure why those commits are "partially signed". However, in terms of merging the repo is configured to automatically squash all commits when merging, which might clear this issue at that point.
Nevertheless, if this causes issues I will look into it and manually fix it.

File: app/src/androidTest/java/tests/SignInTests.kt (hunk 7 above)
Comment (kp-gabriel-sasu) [reply app/src/androidTest/java/tests/SignInTests.kt:40]: That is a good solution only for small cases and it doesn't scale with the solution.

**Pros:**
**Simplicity**: It's straightforward and doesn't require any external data parsing or loading. The data is directly available in the code.
**Performance**: Since it's in-memory, accessing the data is fast. There's no I/O operation involved, as would be the case with reading from a file or database.
**Easy Refactoring**: If you need to change the structure or type of data, refactoring tools in IDEs like Android Studio can easily handle in-code data.

**Cons**:
**Scalability**: For very large datasets, this approach might not be suitable. It can make your test code lengthy and harder to manage.
**Flexibility**: If you want to change the test data, you'd need to change the code and recompile. With external data sources like JSON, you can modify the data without touching the code.
**Version Control**: In-code data can clutter version control diffs if the data changes frequently.
**Reusability**: If multiple tests or test suites need the same data, you'd have to duplicate this data or ensure it's in a shared location, whereas external data sources can be easily shared and reused.

Comment (kp-aldo-pedromingo) [reply app/src/androidTest/java/tests/SignInTests.kt:40]: Completely agree @kp-gabriel-sasu!

I'd suggest these 2 options:
1) We keep the current implementation and we move the test dataset to a .json file
2) We replace the test dataset from a string containing Json to a Kotlin array of `TestData` and keep it inside the source file.

Comment (kp-ruth-ondrada) [reply app/src/androidTest/java/tests/SignInTests.kt:40]: hello @kp-aldo-pedromingo  and @kp-gabriel-sasu ,

Thank you and appreciate your reviews, Maybe we can merge this ticket first and reconsider our approach in the next data driven test, I can create a spike ticket for data driven testing and have a some sort of proposal on the approach.

let me know!

**Comment by kp-ruth-ondrada** [comment]
> > Commits: [a7bf33c](https://github.com/pia-foss/mobile-android/commit/a7bf33c050fb8809260a158e01097280eb6ffeb0) and [d80a119](https://github.com/pia-foss/mobile-android/commit/d80a119df8df61d0eef59011837134c7c38623fb) appear to be partially signed. This might be an issue, as the PR sniffer will block the merge of the PR.
> > @kp-andrei-madescu it looks like your commits are not being correctly signed?
> > A solution for this could be to do a rebase and squash the faulty commits into another.
> 
> Not sure why those commits are "partially signed". However, in terms of merging the repo is configured to automatically squash all commits when merging, which might clear this issue at that point. Nevertheless, if this causes issues I will look into it and manually fix it.

Let me try merging this and see if there's an issue with the partially signed 🙏

File: app/src/androidTest/java/tests/SignInTests.kt (hunk 7 above)
Comment (kp-aldo-pedromingo) [reply app/src/androidTest/java/tests/SignInTests.kt:40]: Cool. Let's do that!

**Review by kp-aldo-pedromingo — APPROVED** [review]
